src/CSnakes.Runtime/Python/PyBuffer.cs

```diff
@@ -0,0 +1,71 @@
+﻿using CSnakes.Runtime.CPython;
+
+namespace CSnakes.Runtime.Python;
+internal class PyBuffer : IPyBuffer
+{
+    private readonly CPythonAPI.Py_buffer _buffer;
+
+    public PyBuffer(PyObject exporter)
+    {
+        _buffer = CPythonAPI.GetBuffer(exporter);
+    }
+
+    public void Dispose()
+    {
+        CPythonAPI.ReleaseBuffer(_buffer);
+    }
+
+    public Int64 Length => _buffer.len;
+
+    public bool Scalar => _buffer.ndim == 0;
+
+    private unsafe void EnsureFormat(char format)
+    {
+        if (!_buffer.format[0].Equals(format))
+        {
+            throw new InvalidOperationException($"Buffer is not a {format}, it is {_buffer.format[0]}");
+        }
+    }
+
+    private void EnsureScalar()
+    {
+        if (!Scalar)
+        {
+            throw new InvalidOperationException("Buffer is not a scalar");
+        }
+    }
+
+    public unsafe Span<Int32> AsInt32Scalar()
```

Comment:
Do you care about endianness of any of these operations? .NET always defers to the platform. I'm assuming CPython does the same.

Reply:
By default, it uses the platform endianness. But, you can specify that as well, I'll try and work out how to test it.

Reply:
I've just put a guard in there for now